egs/iban/s5/local/nnet3/run_tdnn.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# this is the standard "tdnn" system, built in nnet3; it's what we use to
+# call multi-splice.
+
+. cmd.sh
+
+
+# At this script level we don't support not running on GPU, as it would be painfully slow.
+# If you want to run without GPU you'd have to call train_tdnn.sh with --gpu false,
+# --num-threads 16 and --minibatch-size 128.
+
+stage=0
+train_stage=-10
+dir=exp/nnet3/nnet_tdnn_a
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+local/nnet3/run_ivector_common.sh --stage $stage || exit 1;
+
+if [ $stage -le 8 ]; then
+  #  --splice-indexes "-4,-3,-2,-1,0,1,2,3,4  0  -2,2  0  -4,4 0" \
+  #  --num-hidden-layers 5 \
+
+  steps/nnet3/train_tdnn.sh --stage $train_stage \
+    --num-jobs-initial 2 --num-jobs-final 4 \
+    --splice-indexes "-4,-3,-2,-1,0,1,2,3,4  0  -2,2  0  -4,4 0" \
+    --num-epochs 7 \
+    --add-layers-period 1 \
```

Comment:
I haven't actually measured how much the layer-wise discriminative training helps and how long it should be done. But until you test the impact I would recommend `--add-layers-period 2`. If there aren't sufficient iterations for this you could increase the number of epochs.